fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateViewCommand.java

```diff
@@ -0,0 +1,134 @@
+            ShowResultSetMetaData.builder()
+                    .addColumn(new Column("View", ScalarType.createVarchar(20)))
+                    .addColumn(new Column("Create View", ScalarType.createVarchar(30)))
+                    .addColumn(new Column("character_set_client", ScalarType.createVarchar(30)))
+                    .addColumn(new Column("collation_connection", ScalarType.createVarchar(30)))
+                    .build();
+
+    private final TableNameInfo tblNameInfo;
+
+    public ShowCreateViewCommand(TableNameInfo tableNameInfo) {
+        super(PlanType.SHOW_CREATE_VIEW_COMMAND);
+        this.tblNameInfo = tableNameInfo;
+    }
+
+    private void validate(ConnectContext ctx) throws AnalysisException {
+        tblNameInfo.analyze(ctx);
+
+        TableIf tableIf = Env.getCurrentEnv().getCatalogMgr()
+                .getCatalogOrAnalysisException(tblNameInfo.getCtl())
+                .getDbOrAnalysisException(tblNameInfo.getDb()).getTableOrAnalysisException(tblNameInfo.getTbl());
+
+        if (tableIf instanceof MTMV) {
+            ErrorReport.reportAnalysisException("not support async materialized view, "
+                    + "please use `show create materialized view`");
+        }
+
+        PrivPredicate wanted;
+        if (tableIf instanceof View) {
+            wanted = PrivPredicate.SHOW_VIEW;
+        } else {
+            wanted = PrivPredicate.SHOW;
+        }
+
+        if (!Env.getCurrentEnv().getAccessManager().checkTblPriv(ConnectContext.get(),
+                tblNameInfo.getCtl(), tblNameInfo.getDb(), tblNameInfo.getTbl(), wanted)) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_TABLEACCESS_DENIED_ERROR, "SHOW CREATE TABLE",
+                                                ConnectContext.get().getQualifiedUser(),
+                                                ConnectContext.get().getRemoteIP(),
+                                                tblNameInfo.getDb() + ": " + tblNameInfo.getTbl());
+        }
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitShowCreateViewCommand(this, context);
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        // If dbName is not provided, use the default database from the context
+        String effectiveDbName = (tblNameInfo.getDb() == null || tblNameInfo.getDb().isEmpty())
+                        ? ctx.getDatabase() : tblNameInfo.getDb();
+
+        // If catalog is not provided, use the default catalog (e.g., internal)
+        String effectiveCatalog = (tblNameInfo.getCtl() == null || tblNameInfo.getCtl().isEmpty())
+                        ? "internal" : tblNameInfo.getCtl();
+
+        // Fetch the catalog, database, and view metadata
+        Env env = Env.getCurrentEnv();
+        DatabaseIf db = env.getCatalogMgr().getCatalogOrAnalysisException(effectiveCatalog)
```

Comment:
keep behavior same as old code:
```suggestion
DatabaseIf db = ctx.getEnv().getCatalogMgr().getCatalogOrAnalysisException(tblNameInfo.getCtl())
                .getDbOrAnalysisException(tblNameInfo.getDb());
```

Reply:
done - please review